modules/crop-workflowoperation/src/main/java/org/opencastproject/workflow/handler/crop/CropWorkflowOperationHandler.java

```diff
@@ -0,0 +1,189 @@
+  }
+
+  /** The composer service */
+  private CropService cropService = null;
+
+  /** The local workspace */
+  private Workspace workspace = null;
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowOperationHandler#getConfigurationOptions()
+   */
+  @Override
+  public SortedMap<String, String> getConfigurationOptions() {
+    return CONFIG_OPTIONS;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance, JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext jobContext)
+          throws WorkflowOperationException {
+    logger.debug("Running cropping on workflow {}", workflowInstance.getId());
+
+    final String targetTrackId = idBuilder.createNew().toString();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+
+    String trackFlavor = StringUtils.trimToNull(operation.getConfiguration(PROP_ANALYSIS_TRACK_FLAVOR));
+    List<String> targetTags = asList(operation.getConfiguration(PROP_TARGET_TAGS));
+    List<Track> candidates = new ArrayList<>();
+    if (trackFlavor != null) {
+      candidates.addAll(Arrays.asList(mediaPackage.getTracks(MediaPackageElementFlavor.parseFlavor(trackFlavor))));
+    } else {
+      candidates.addAll(Arrays.asList(mediaPackage.getTracks(MediaPackageElements.PRESENTATION_SOURCE)));
+    }
+
+    Iterator<Track> ti = candidates.iterator();
+    while (ti.hasNext()) {
+      Track t = ti.next();
+      if (!t.hasVideo()) {
+        ti.remove();
+      }
+    }
+
+    if (candidates.size() == 0) {
+      logger.info("No matching tracks available for cropping in workflow {}", workflowInstance);
+      return createResult(WorkflowOperationResult.Action.CONTINUE);
+    }
+
+    if (candidates.size() > 1) {
+      logger.info("Found more than one track to crop");
+    }
+
+    // all canidates are cropped in the following loop, but I'm not sure if I'm doing this in the right way
+    Job job = null;
+    for (int i = 0; i < candidates.size(); i++) {
+      Track track = candidates.get(0);
+
+      Track croppedTrack = null;
+
+      try {
+        job = cropService.crop(track);
+        if (!waitForStatus(job).isSuccess()) {
+          throw new WorkflowOperationException("Video cropping of " + track + " failed");
+        }
+
+        croppedTrack = (Track) MediaPackageElementParser.getFromXml(job.getPayload());
+        mediaPackage.add(croppedTrack);
+        croppedTrack.setIdentifier(targetTrackId);
+        croppedTrack.setURI(workspace
+                .moveTo(croppedTrack.getURI(), mediaPackage.getIdentifier().toString(), croppedTrack.getIdentifier(),
+                        croppedTrack.getIdentifier() + "corpped"));
+        croppedTrack.setMimeType(track.getMimeType());
```

Comment:
The output mime type could be completely different from the input mime type. If you want it to be set, you can use inspection on the output track.